pkg/controller/master/virtualmachine/vmi_network_controller.go

```diff
@@ -85,3 +98,52 @@ func (h *VMNetworkController) updateVMDefaultNetworkMacAddress(vmi *kubevirtv1.V
 
 	return nil
 }
+
+func (h *VMNetworkController) updateVMMacAddressFromAnnotation(vmi *kubevirtv1.VirtualMachineInstance) error {
+	vm, err := h.vmCache.Get(vmi.Namespace, vmi.Name)
+	if err != nil {
+		return err
+	}
+
```

Comment:
we should have a check if vm has a deletion timestamp set, in which we can return immediately.

This will handle the case when a VMI deletion is triggered due to deletion of the VM

Reply:
Sounds good. Addressed in https://github.com/harvester/harvester/pull/9395/commits/02522d77f047db1a6a7501277c095897aedda353